kortex_driver/src/non-generated/driver/kortex_arm_driver.cpp

```diff
@@ -526,6 +527,69 @@ void KortexArmDriver::startActionServers()
     }
 }
 
+void KortexArmDriver::moveArmWithinJointLimits()
+{
+    auto joint_angles = m_base->GetMeasuredJointAngles();
+
+    std::map<int, float> limited_joints;
+    if (m_degrees_of_freedom == 6)
+    {
+        // We add angle limitations for joints 1,2 and 4 on 6 dof
+        limited_joints[1] = 128.9;
+        limited_joints[2] = 147.8;
+        limited_joints[4] = 120.3;
+    } 
+    else 
+    {
+        // We add angle limitations for joints 1,3 and 5 on 7 dof
+        limited_joints[1] = 128.9;
+        limited_joints[3] = 147.8;
+        limited_joints[5] = 120.3;
+    }
+
+    Kinova::Api::Base::JointSpeeds joint_speeds = Kinova::Api::Base::JointSpeeds();
+    Kinova::Api::Base::JointSpeed* joint_speed = joint_speeds.add_joint_speeds();
+
+    static const int TIME_COMPENSATION = 100;
+    static const int DEFAULT_JOINT_SPEED = 10;
+    static const int TIME_SPEED_RATIO = 1000 / DEFAULT_JOINT_SPEED;
+
+    float angle;
+    for (unsigned int i = 0; i < m_degrees_of_freedom; i++)
+    {
+        angle = joint_angles.joint_angles(i).value();
+
+        // Angles received by GetMeasuredJointAngles are in range [0,360], but waypoints are in range [-180, 180]
+        angle = m_math_util.toDeg(m_math_util.wrapRadiansFromMinusPiToPi(m_math_util.toRad(angle)));
+
+        if(limited_joints.find(i) != limited_joints.end())
+        {
+            float delta = m_math_util.findDeltaFromBoundaries(angle, limited_joints.at(i));
```

Comment:
this function returns a `double` and you're casting to float, should be double everywhere or float everywhere